pkg/server/backuprestoreserver.go

```diff
@@ -220,18 +220,13 @@ func (b *BackupRestoreServer) runEtcdProbeLoopWithSnapshotter(ctx context.Contex
 				// If the previous full snapshot doesn't exist or is not marked as final, take a final full snapshot
 				if ssr.PrevFullSnapshot == nil || !ssr.PrevFullSnapshot.IsFinal {
 					b.logger.Infof("Taking final full snapshot...")
-					var snapshot *brtypes.Snapshot
-					if snapshot, err = ssr.TakeFullSnapshotAndResetTimer(true); err != nil {
+					leaseUpdatectx, cancel := context.WithTimeout(ctx, brtypes.LeaseUpdateTimeoutDuration)
+					defer cancel()
```

Comment:
If you create a very restricted context already here, then it is probably already expired when the update happens because the full snapshot can take a while. I think you can drop this context creation since you already create another on in `TakeFullSnapshotAndResetTimer`.